test/compound_and_array_test.mjs

```diff
@@ -9,24 +9,57 @@ async function compound_array_test() {
 
   assert.deepEqual(
     f.get('float_arr').value,
-    new Float64Array([0, 1, 2, 3, 4, 5, 6, 7])
+    [
+      [
+        [0,1],
+        [2,3]
+      ],
+      [
+        [4,5],
+        [6,7]
+      ]
+    ]
   );
 
-  assert.deepEqual(f.get('string_arr').value, [
-    'hello',
-    'there',
-    'hello',
-    'there',
-    'hello',
-    'there',
-    'hello',
-    'there',
-  ]);
+  assert.deepEqual(
+    f.get('string_arr').value,
+    [
+      [
+        ["hello","there"],
+        ["hello","there"]
+      ],
+      [
+        ["hello","there"],
+        ["hello","there"]
+      ]
+    ]
+  );
 
-  assert.deepEqual(f.get('compound').value, [
-    [new Float64Array([0, 1, 2, 3]), ['hello', 'there', 'hello', 'there']],
-    [new Float64Array([4, 5, 6, 7]), ['hello', 'there', 'hello', 'there']],
-  ]);
+  assert.deepEqual(
+    f.get('compound').value,
+    [
+      [
+        [
+          [0,1],
+          [2,3]
+        ],
+        [
+          ["hello","there"],
+          ["hello","there"]
+        ]
+      ],
+      [
+        [
+          [4,5],
+          [6,7]
+        ],
+        [
+          ["hello","there"],
+          ["hello","there"]
+        ]
+      ]
+    ]
+  );
```

Comment:
If I understand correctly, this gets converted to JSON compatible output as the `compound` dataset includes fields with array types ?

Other compound types (such as complex) will still have the same behaviour ? If so, it might be worthwhile to add a compound dataset with simple numeric fields (to test the compound type independently from the array type)

Reply:
Only the ARRAY types get converted automatically... you can still have a COMPOUND type with non-json-compatible ```value``` if not all members are in an ARRAY.